lib/AppHelper.php

```diff
@@ -468,7 +465,7 @@ public static function waitFor($test, $timeout=60, $interval=1) {
 		do {
 			$lastTry = time();
 			if($test()) return true;
-			while(time() < $lastTry+$interval) sleep(1);
+			while(time() < $lastTry+$interval && time() < $startTime+$timeout) sleep(1);
```

Comment:
Well, I tuned it taking into account your suggestions

Reply:
OK. This function is quite interesting but a bit tricky to describe in human language. For example, even if the code snippet I suggested above is a valid implementation of `waitFor()`, it probably deviates a bit from the current documentation because similarly to your previous implementation, I treated `$timeout` as the maximum wait duration during each wait cycle (to resolve `$test()` to a truthy value), instead of treating the parameter as the maximum wait duration since `waitFor()` is called (which you seem to aim for in the latest implementation of the function).

I haven't really checked the new implementation, but I would say that variable `$timeoutPassed = $startTime+$timeout;` is probably better called `$timeoutLimit` and `sleep(min($timeoutPassed - time(), $lastTry+$interval - time()));` does not exclude a negative value being passed as argument to the `sleep()` function (which would throw an exception unexpectedly)...

Reply:
`sleep(max(0, min($timeoutPassed - time(), $lastTry+$interval - time())));` ? :-D